PWM_Pins.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+// The interface for PWM pin controllers
+
+class PWM_Pins {
+  // don't want to make abstract, because virtual may not optimize away to inlines
```

Comment:
But, I can't express the idea of interface right. sad. so, this is mostly just a place holder/explainer. This class is never actually used. No code in it is currently used.